.hooks/typechecking_installer.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+# Usage
+# To install the typechecking linters, simply run the script in your terminal:
+
+# sh
+# This will install the linters and set up the pre-commit hook to run them on each changed Python file.
+# -o, --post-commit, -p, --pre-commit, default -o post commit install
+
+# Requirements
+# This script requires the following tools to be installed:
+
+# Git
+# Black
+# Ruff
+# Mypy
+# Make sure these tools are installed and available in your system's PATH before running the script.
+
+# Define the hook script
+HOOK_SCRIPT="#!/bin/bash
+# Get the list of changed Python files in the darwin/future folder
+FILES=\$(git diff --diff-filter=MA --name-only master | grep 'darwin/future/.*\.py$')
+if [ -z \"\$FILES\" ]; then
+    exit 0
+fi
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+echo Typechecking Hook
+echo ----------------------------------------
+echo checking \$FILES
+# Run the linters on each changed file
+echo -e '\nRunning Black'
+echo ----------------------------------------
+BLACK_FAILED=0
+black --check \$FILES || BLACK_FAILED=1
```

Comment:
These could each be like

```bash
BLACK_FAILED=`black --check \$FILES`
```

and so on, with the same function, but again, if it ain't broke.